src/Middleware/CORS/src/Infrastructure/CorsOptions.cs

```diff
@@ -12,22 +13,16 @@ namespace Microsoft.AspNetCore.Cors.Infrastructure
     public class CorsOptions
     {
         private string _defaultPolicyName = "__DefaultCorsPolicy";
-        private IDictionary<string, CorsPolicy> PolicyMap { get; } = new Dictionary<string, CorsPolicy>();
+
+        internal IDictionary<string, (CorsPolicy policy, Task<CorsPolicy> policyTask)> PolicyMap { get; }
```

Comment:
Is the tuple to avoid accessing Task.Result? The task will always be complete so I don't think it is a big deal.

A comment explaining why a task is cached would be useful